app/models/user.rb

```diff
@@ -63,6 +67,27 @@ def is_application?
     false
   end
 
+  # State helpers.
+  #
+  # A User model begins life in the "temp" state, and can then be claimed by another user
+  # who originated from an OAuth login. Upon it being claimed it will be removed and it's
+  # data merged with the claimant.
+  #
+  # A User can also be created by a one of the consumer applications as a stand-in
+  # for a person who has not yet (or may never) created an account.  In this case
+  # the User model will be in the "unclaimed" state.  When the User does signup, they
```

Comment:
@nathanstitt did you address this case?  If not, can you add a spec (maybe in the sessions callback spec, which already has some Google/FB stuff) to show that this fails and then fix?